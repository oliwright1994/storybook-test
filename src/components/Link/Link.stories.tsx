import { Meta } from '@storybook/react/types-6-0'
import React from 'react'

import Link, { ILink } from './Link'

export default { component: Link, title: 'Components/Link' } as Meta
export const Standard = (args: ILink): JSX.Element => <Link {...args} />
Standard.args = {
  children: 'link',
  isExternal: false,
  isDisabled: false,
  as: '',
  href: '/',
}
