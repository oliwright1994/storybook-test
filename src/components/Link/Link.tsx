import { Link as ChakraLink, LinkProps as IChakraLink } from '@chakra-ui/core'
import React, { FC } from 'react'

export interface ILink extends Pick<IChakraLink, 'isExternal' | 'isDisabled' | 'onClick' | 'as'> {
  href?: string
}

const Link: FC<ILink> = ({ onClick, href, isExternal, isDisabled, children, as }) => (
  <ChakraLink onClick={onClick} href={href} isExternal={isExternal} isDisabled={isDisabled} as={as}>
    {children}
  </ChakraLink>
)

export default Link
