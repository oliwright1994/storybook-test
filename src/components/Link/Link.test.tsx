import { fireEvent } from '@testing-library/react'
import React from 'react'
import { renderWithBrightTheme } from 'testing/helpers/themeHelper'

import Link from '.'

describe('<Link />', () => {
  const props = {
    href: 'test/url',
    children: 'Link text',
  }

  it('should render as expected', () => {
    const { asFragment } = renderWithBrightTheme(<Link {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should display text', () => {
    const { container } = renderWithBrightTheme(<Link {...props} />)

    expect(container).toHaveTextContent(props.children)
  })

  it('should add disabled attribute', () => {
    const { getByText } = renderWithBrightTheme(<Link {...props} isDisabled />)

    expect(getByText(/link/i)).toHaveAttribute('aria-disabled')
  })

  it('should fire onClick', () => {
    const clickHandler = jest.fn()
    const { getByText } = renderWithBrightTheme(<Link {...props} onClick={clickHandler} />)

    fireEvent.click(getByText(/link/i))
    expect(clickHandler).toHaveBeenCalled()
  })
})
