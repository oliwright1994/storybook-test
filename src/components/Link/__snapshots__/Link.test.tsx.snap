// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should render as expected 1`] = `
<DocumentFragment>
  <div
    class="css-u5zpo1"
    href="test/url"
  >
    Link text
  </div>
</DocumentFragment>
`;
